### PR TITLE
Windows Installer [API-1446]

### PR DESCRIPTION
**Pull request by yuce**
This PR adds the initial Windows installer script.
Inno Setup 6 is required to build the installer: https://jrsoftware.org/isdl.php

After adding the path that contains `ISSC.exe` (typically: C:\Program Files (x86)\Inno Setup 6`) run `make installer` to create the installer in the project root.

### CODE REVIEW

File: extras/windows/hazelcast-clc-installer.iss [hunk 1]
```diff
@@ -0,0 +1,103 @@
+#define MyAppName "Hazelcast CLC"
+#define MyAppVersion "v5.2.0-beta2"
+#define MyAppPublisher "Hazelcast, Inc."
+#define MyAppURL "https://www.hazelcast.com/"
```
Comment (utku-caglayan) [inline extras/windows/hazelcast-clc-installer.iss:4]: Maybe we can point users to the GitHub URL here. Wdyt?

File: extras/windows/pre.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Welcome to the Hazelcast CLC Install Wizard.
```
Comment (utku-caglayan) [inline extras/windows/pre.txt:1]:
```suggestion
Welcome to the Hazelcast CLC Installation Wizard.
```

File: make.cmd [hunk 3]
```diff
@@ -22,7 +22,14 @@ if errorlevel 1 (
 goto :end
 
 :build
+    go-winres make --product-version=%CLC_VERSION% --file-version=%CLC_VERSION%
+    go generate ./...
```
Comment (utku-caglayan) [inline make.cmd:26]: Is this step necessary?

**Review by utku-caglayan — COMMENTED** [review]
Thank you for such critical work!

File: extras/windows/hazelcast-clc-installer.iss (hunk 1 above)
Comment (yuce) [reply extras/windows/hazelcast-clc-installer.iss:4]: Let's discuss that after the release.

File: make.cmd (hunk 3 above)
Comment (yuce) [reply make.cmd:26]: I guess? Since it adds the Windows releated stuff.

File: extras/windows/pre.txt (hunk 2 above)
Comment (yuce) [reply extras/windows/pre.txt:1]: It seems both the "Install Wizard" and "Installation Wizard" are used, as far as I've checked on the internet.

**Review by utku-caglayan — APPROVED** [review]

**Pull request merged**